examples/dynamic_api.rs

```diff
@@ -19,59 +19,41 @@ async fn main() -> anyhow::Result<()> {
 
     let ns_filter = std::env::var("NAMESPACE").ok();
 
-    // The following loops turn the /api or /apis listers into kube::api::Resource
-    // objects which can be used to make dynamic api calls.
-    // This is slightly awkward because of corev1 types
-    // and data split over the list types and the inner get calls.
-
-    // loop over all api groups (except core v1)
-    let apigroups = client.list_api_groups().await?;
-    for g in apigroups.groups {
-        warn!("api group: {}", g.name);
-        let ver = g
-            .preferred_version
-            .as_ref()
-            .or_else(|| g.versions.first())
-            .expect("preferred or versions exists");
-        let apis = client.list_api_group_resources(&ver.group_version).await?;
-        print_group(&client, &ver.group_version, apis, ns_filter.as_deref()).await?;
-    }
-
-    warn!("core/v1 legacy group");
-    let coreapis = client.list_core_api_versions().await?;
-    assert_eq!(coreapis.versions.len(), 1);
-    let corev1 = client.list_core_api_resources(&coreapis.versions[0]).await?;
-    print_group(&client, &coreapis.versions[0], corev1, ns_filter.as_deref()).await
-}
-
-async fn print_group(
-    client: &Client,
-    group_version: &str,
-    apis: APIResourceList,
-    ns_filter: Option<&str>,
-) -> anyhow::Result<()> {
-    for ar in apis.resources {
-        if !ar.verbs.contains(&"list".to_string()) {
-            continue;
-        }
-        let gvk = GroupVersionKind::from_api_resource(&ar, &apis.group_version);
-        let api: Api<DynamicObject> = if ar.namespaced {
-            if let Some(ns) = ns_filter {
-                Api::namespaced_with(client.clone(), ns, &gvk)
+    let discovery = Discovery::new(&client).await?;
```

Comment:
I do like the idea of a complete `Discovery` helper, but it would be nice to be able to hit the specific subset of the api as well.

There are quite a lot of queries we can make here: https://docs.rs/k8s-openapi/0.11.0/k8s_openapi/fn.get_batch_v1_api_resources.html?search=api_resources

Reply:
> https://docs.rs/k8s-openapi/0.11.0/k8s_openapi/fn.get_batch_v1_api_resources.html?search=api_resources

IIUC, all these methods can be replaced by a single [`Client::list_api_group_resources`](https://docs.rs/kube/0.52.0/kube/struct.Client.html#method.list_api_group_resources).

I agree it can be useful to create a `Discovery` that contains only a subset of API groups, but I propose leaving it as future work.

Reply:
Ok, that's fine. Maybe leave a TODO in the code for it for now :-)